src/Symfony/Component/Translation/Provider/ProviderInterface.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Provider;
+
+use Symfony\Component\Translation\TranslatorBag;
+
+/**
+ * Providers are used to sync translations with a translation provider.
+ */
+interface ProviderInterface
+{
+    /**
+     * Returns the Provider name.
+     */
+    public function getName(): string;
+
+    /**
+     * Writes given translations to the provider.
+     *
+     * * Translations available in the MessageCatalogue only must be created.
+     * * Translations available in both the MessageCatalogue and on the provider
+     * must be overwritten.
+     * * Translations available on the provider only must be kept.
+     */
+    public function write(TranslatorBag $translatorBag): void;
```

Comment:
No, because we call `getDomains` which is not in the `TranslatorBagInterface`

Reply:
Since `TranslatorBag` is `final`, this creates a hard coupling between the interface and the implementation. This looks like an abstraction leak to me. Can't we really do something with `TranslatorBagInterface`? Otherwise, we should at least make the class not final, or add an interface to provide the missing piece of abstraction.